src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/VenueAddressReaderService.java

```
@@ -0,0 +1,78 @@
+
+        if (CollectionUtils.isEmpty(venueAddressList)) {
+            throw new VenueAddressReaderException("Venue address list not found for " + managingOffice);
+        }
+
+        VenueAddress venueAddressItem = venueAddressList.stream()
+                .filter(v -> v.getVenue().equals(hearingVenue))
+                .findAny()
+                .orElse(null);
+
+        return venueAddressItem != null
+                ? StringUtils.defaultIfBlank(venueAddressItem.getAddress(), hearingVenue)
+                : hearingVenue;
+    }
+
+    private String getHearingVenue(HearingType hearingType, String caseTypeId) {
+        if (ENGLANDWALES_CASE_TYPE_ID.equals(caseTypeId)) {
+            return hearingType.getHearingVenue().getValue().getLabel();
+        } else if (SCOTLAND_CASE_TYPE_ID.equals(caseTypeId)) {
+            return getHearingVenueScotland(hearingType);
+        } else {
+            throw new IllegalArgumentException("Unexpected case type id " + caseTypeId);
+        }
+    }
+
+    private String getHearingVenueScotland(HearingType hearingType) {
+        var venue = hearingType.getHearingVenueScotland();
+        if (StringUtils.isBlank(venue)) {
+            log.error("No office found for hearing {}", hearingType.getHearingNumber());
+            return null;
+        }
+
+        switch (venue) {
+            case GLASGOW_OFFICE:
+                return hearingType.getHearingGlasgow().getSelectedLabel();
+            case ABERDEEN_OFFICE:
+                return hearingType.getHearingAberdeen().getSelectedLabel();
+            case DUNDEE_OFFICE:
+                return hearingType.getHearingDundee().getSelectedLabel();
+            case EDINBURGH_OFFICE:
+                return hearingType.getHearingEdinburgh().getSelectedLabel();
+            default:
+                log.error("No {} venue found", venue);
+                return null;
```

I think it would be better to throw an exception rather than return null